common/core/nesteddisplay.cpp

```diff
@@ -22,16 +22,145 @@
+  socklen_t clilen;
+  struct sockaddr_in client_addr;
+
+  if (sock_fd >= 0) {
+    clilen = sizeof(client_addr);
+    *connected = false;
+    *client_sock_fd = accept(sock_fd, (struct sockaddr *)&client_addr, &clilen);
+    mEnabled = false;
+    *connected = true;
+  }
+}
+#endif
+
+NestedDisplay::NestedDisplay(uint32_t gpu_fd,
+                             NativeBufferHandler *buffer_handler) {
+#ifdef NESTED_DISPLAY_SUPPORT
+  int ret;
+  struct ioctl_hyper_dmabuf_tx_ch_setup msg;
+  memset(&msg, 0, sizeof(ioctl_hyper_dmabuf_tx_ch_setup));
+
+  resource_manager_.reset(new ResourceManager(buffer_handler));
+  if (!resource_manager_) {
+    ETRACE("Failed to construct hwc layer buffer manager");
+  }
+  compositor_.Init(resource_manager_.get(), gpu_fd);
+
+  mHyperDmaBuf_Fd = open(HYPER_DMABUF_PATH, O_RDWR);
+  if (mHyperDmaBuf_Fd < 0)
+    ETRACE("Hyper DmaBuf: open hyper dmabuf device node %s failed because %s",
+           HYPER_DMABUF_PATH, strerror(errno));
+  else {
+    ETRACE("Hyper DmaBuf: open hyper dmabuf device node %s successfully!",
+           HYPER_DMABUF_PATH);
+    /* TODO: add config option to specify which domains should be used, for now
+     * we share always with dom0 */
+    msg.remote_domain = 0;
+    ret = ioctl(mHyperDmaBuf_Fd, IOCTL_HYPER_DMABUF_TX_CH_SETUP, &msg);
+    if (ret) {
+      ETRACE(
+          "Hyper DmaBuf: IOCTL_HYPER_DMABUF_TX_CH_SETUP failed with error %d\n",
+          ret);
+      close(mHyperDmaBuf_Fd);
+      mHyperDmaBuf_Fd = -1;
+    } else
+      ETRACE("Hyper DmaBuf: IOCTL_HYPER_DMABUF_TX_CH_SETUP Done!\n");
+  }
+#endif
 }
 
 NestedDisplay::~NestedDisplay() {
+#ifdef NESTED_DISPLAY_SUPPORT
+  if (mHyperDmaBuf_Fd > 0) {
+    for (auto it = mHyperDmaExportedBuffers.begin();
+         it != mHyperDmaExportedBuffers.end(); ++it) {
+      struct ioctl_hyper_dmabuf_unexport msg;
+      int ret;
+      msg.hid = it->second.hyper_dmabuf_id;
+      // Todo: hyper dmabuf free delay is fixed to 1s now!
+      msg.delay_ms = 1000;
+      ret = ioctl(mHyperDmaBuf_Fd, IOCTL_HYPER_DMABUF_UNEXPORT, &msg);
+      if (ret) {
+        ETRACE(
+            "Hyper DmaBuf: IOCTL_HYPER_DMABUF_UNEXPORT ioctl failed %d "
+            "[0x%x]\n",
+            ret, it->second.hyper_dmabuf_id.id);
+      } else {
+        ETRACE("Hyper DmaBuf: IOCTL_HYPER_DMABUF_UNEXPORT ioctl Done [0x%x]!\n",
+               it->second.hyper_dmabuf_id.id);
+        mHyperDmaExportedBuffers.erase(it);
+      }
+    }
+
+    close(mHyperDmaBuf_Fd);
+    mHyperDmaBuf_Fd = -1;
+  }
+
+  if (mclient_sock_fd >= 0)
+    close(mclient_sock_fd);
+
+  if (msock_fd >= 0)
+    close(msock_fd);
+
+  resource_manager_->PurgeBuffer();
+  compositor_.Reset();
+#endif
 }
 
-void NestedDisplay::InitNestedDisplay() {
+void NestedDisplay::InitNestedDisplay(uint32_t width, uint32_t height,
+                                      uint32_t port) {
+  width_ = width;
+  height_ = height;
+  port_ = port;
+
+#ifdef NESTED_DISPLAY_SUPPORT
```

Comment:
You can remove this and have this check in caller side i.e in IAHWC that it calls only if the macro is defined